sdk/core/Azure.Core/src/Shared/ClientDiagnostics.cs

```diff
@@ -31,16 +31,31 @@ internal sealed class ClientDiagnostics: DiagnosticScopeFactory
                 options.Diagnostics.LoggedHeaderNames.ToArray());
         }
 
+        /// <summary>
+        /// Partial method that can optionally be defined to extract the error
+        /// message, code, and details in a service specific manner.
+        /// </summary>
+        /// <param name="content">The error content.</param>
+        /// <param name="message">The error message.</param>
+        /// <param name="errorCode">The error code.</param>
+        /// <param name="additionalInfo">Additional error details.</param>
+        partial void ExtractFailureContent(
```

Comment:
This signature looks very unusual, should we make this method virtual?

Reply:
I was looking for something with zero cost to everyone else who didn't need it and trying to be minimally invasive to code I don't own.  I'm happy to change this to whatever you'd prefer it to be.

Reply:
Lets do virtual, I think JIT can optimize this call away.

Reply:
We chatted offline about this and decided to stay with what I have here for the short term.  It's currently a sealed type and we'll rethink the approach for customizing error messages as the distributed tracing plans evolve.